### PR TITLE
not report dependency if path is invalid

### PR DESCRIPTION
**Pull request by v-pegao**

### CODE REVIEW

File: src/Microsoft.DocAsCode.Dfm/CodeRenderer/DfmCodeRenderer.cs [hunk 1]
```diff
@@ -39,27 +40,38 @@ public virtual StringBuffer Render(IMarkdownRenderer renderer, DfmFencesToken to
 
             try
             {
-                // Always report original dependency
-                context.ReportDependency(token.Path);
```
Comment (superyyrrzz) [inline src/Microsoft.DocAsCode.Dfm/CodeRenderer/DfmCodeRenderer.cs]: What about keep reportDependency here, but wrap this with `isValidPath`?

File: src/Microsoft.DocAsCode.Dfm/CodeRenderer/DfmCodeRenderer.cs [hunk 2]
```diff
@@ -39,27 +40,38 @@ public virtual StringBuffer Render(IMarkdownRenderer renderer, DfmFencesToken to
...
             {
+                if (isValidPath(token.Path))
+                {
+                    context.ReportDependency(token.Path);
+                }
                 return RenderReferenceNotFoundErrorMessage(renderer, token);
             }
         }
 
+        public bool isValidPath(string path) => !_invalidChars.Any(c => path.Contains(c));
```
Comment (superyyrrzz) [inline src/Microsoft.DocAsCode.Dfm/CodeRenderer/DfmCodeRenderer.cs]: Can we reuse `PathUtility.IsValidFilePath`?

File: src/Microsoft.DocAsCode.Dfm/CodeRenderer/DfmCodeRenderer.cs (hunk 1 above)
Comment (superyyrrzz) [reply src/Microsoft.DocAsCode.Dfm/CodeRenderer/DfmCodeRenderer.cs]: // Always report original dependency **when path is valid**

File: src/Microsoft.DocAsCode.Dfm/CodeRenderer/DfmCodeRenderer.cs (hunk 2 above)
Comment (v-pegao) [reply src/Microsoft.DocAsCode.Dfm/CodeRenderer/DfmCodeRenderer.cs]: As I tested, it's working well

**Comment by superyyrrzz** [comment]
cc @TianqiZhang

**Pull request merged**